irods/test/connection_test.py

```diff
@@ -65,6 +67,47 @@ def test_reply_failure(self):
             with self.assertRaises(NetworkException):
                 conn.reply(0)
 
+    def test_that_connection_timeout_works__issue_377(self):
+        sess = self.sess
+
+        # skip if server is remote
+        if sess.host not in ('localhost', socket.gethostname()):
+            self.skipTest('Requires access to server-side file(s)')
+        r = sess.resources.create('hello', 'unixfilesystem', host='localhost')
+        rand = os.urandom(1024)*64
+
+        # Create a large file.
+        size = 1024**3*4
+        physical_path = '/tmp/testfile'
```

Comment:
is this windows-friendly?

Reply:
I have changed to using `NamedTemporaryFile` , and a PUT instead of a REG in the test. so I think we are Windows friendly now.